### ISSUE TITLE
Running az commands in parallel on a CI server triggers race condition in Knack's ensure_dir function

### ISSUE THREAD

**Issue by timstallmann** [Azure/azure-cli#9825]
> ### `az feedback` auto-generates most of the information requested below, as of CLI version 2.0.62

**Describe the bug**
The `ensure_dir` command in Knack config, which is called by Azure's CLI code, fails to catch an `EEXISTS` error which is triggered in the event of a race condition when multiple parallel threads each try to create the same directory.

I've reported this bug also in the Knack repository (see https://github.com/microsoft/knack/issues/158) and have a pending PR for it there but flagging here as well since this repository seem to have more activity.

Sample output:
```

The command failed with an unexpected error. Here is the traceback:

The command failed with an unexpected error. Here is the traceback:

[Errno 17] File exists: '/root/.azure/commands'
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/knack/cli.py", line 206, in invoke
    cmd_result = self.invocation.execute(args)
  File "/usr/local/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 447, in execute
    load_cmd_tbl_func=self.commands_loader.load_command_table, args=args)
  File "/usr/local/lib/python3.6/site-packages/knack/cli.py", line 162, in raise_event
    func(self, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/azure/cli/core/azlogging.py", line 80, in init_command_file_logging
    self._init_command_logfile_handlers(cmd_logger, args)  # pylint: disable=protected-access
  File "/usr/local/lib/python3.6/site-packages/azure/cli/core/azlogging.py", line 87, in _init_command_logfile_handlers
    ensure_dir(self.command_log_dir)
  File "/usr/local/lib/python3.6/site-packages/knack/util.py", line 103, in ensure_dir
    os.makedirs(d)
  File "/usr/local/lib/python3.6/os.py", line 220, in makedirs
    mkdir(name, mode)
FileExistsError: [Errno 17] File exists: '/root/.azure/commands'[Errno 17] File exists: '/root/.azure/commands'
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/knack/cli.py", line 206, in invoke
    cmd_result = self.invocation.execute(args)
  File "/usr/local/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 447, in execute
    load_cmd_tbl_func=self.commands_loader.load_command_table, args=args)
  File "/usr/local/lib/python3.6/site-packages/knack/cli.py", line 162, in raise_event
    func(self, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/azure/cli/core/azlogging.py", line 80, in init_command_file_logging
    self._init_command_logfile_handlers(cmd_logger, args)  # pylint: disable=protected-access
  File "/usr/local/lib/python3.6/site-packages/azure/cli/core/azlogging.py", line 87, in _init_command_logfile_handlers
    ensure_dir(self.command_log_dir)
  File "/usr/local/lib/python3.6/site-packages/knack/util.py", line 103, in ensure_dir
    os.makedirs(d)
  File "/usr/local/lib/python3.6/os.py", line 220, in makedirs
    mkdir(name, mode)
FileExistsError: [Errno 17] File exists: '/root/.azure/commands'
```

**To Reproduce**
Run multiple copies of `az storage blob upload-batch` in parallel on a newly-provisioned server (e.g. CircleCI). Code sample:
```
for extension in extensions:
    try:
        mime = mimetypes.types_map[extension]
    except KeyError:
        print('MIME Type not found: %s' % extension)
        continue
    command = [
        "az", 
        "storage", 
        "blob", 
        "upload-batch", 
        "--connection-string",
        os.getenv('AZURE_STORAGE_CONNECTION_STRING'),"-s", 
        "public", 
        "-d", 
        os.getenv('AZURE_DESTINATION_BLOB'), 
        "--pattern", 
        "*"+extension, 
        "--content-type", 
        mime
    ]
    commands.append(command)

workers = min(MAX_WORKERS, len(commands))
with futures.ThreadPoolExecutor(workers) as executor:
    res = executor.map(run_command, commands)
```

**Expected behavior**
Azure cli should not throw an exception here.

**Environment summary**
```
Linux-4.15.0-1035-aws-x86_64-with
Python 3.6.4
Shell: bash

azure-cli 2.0.67
```

**Additional context**

**Comment by tjprescott** [comment]
Closing in favor of the knack issue.
